src/main/kotlin/com/sourcegraph/cody/edit/EditUtil.kt

```diff
@@ -91,4 +100,28 @@ object EditUtil {
       color.darker()
     }
   }
+
+  /**
+   * Return true if the currently selected text editor for `project` is visiting a file that is
+   * currently ignored by CodyIgnore.
+   */
+  fun currentlySelectedFileIsIgnored(project: Project?): Boolean {
```

Comment:
First, is this function used? Cmd-F doesn't find it (but GitHub code review sucks for cmd-F...)

Second, let's de-dup this with `IgnoreOracle.getInstance(project).isEditingIgnoredFile`, which the autocomplete status indicator in the status bar is built on.

That does not block, it just gives you whether the last focused file is ignored, with some some crude handling of races if the focus changes quickly (the `focusedPolicy != policy && willFocusUri == uri` part.) (Crude because the reads of those two volatile fields could be out of sync...)